cdap-standalone/src/main/java/co/cask/cdap/StandaloneMain.java

```diff
@@ -200,6 +214,32 @@ public void startUp() throws Exception {
     cleanupTempDir();
 
     ConfigurationLogger.logImportantConfig(cConf);
+    if (auditEnabled) {
+      String separator = System.getProperty("file.separator");
+      String classpath = System.getProperty("java.class.path");
+      String javaPath = System.getProperty("java.home") + separator + "bin" + separator + "java";
+
+      File zkLogs = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR), "zk-logs");
+      String zkPort = cConf.get(Constants.CFG_LOCAL_ZOOKEEPER_SERVER_PORT);
+      LOG.info(String.format("Zookeeper server classpath : %s", classpath));
+      ProcessBuilder zkProcessBuilder = new ProcessBuilder(javaPath, "-cp", classpath,
+                                                           ZooKeeperServerMain.class.getCanonicalName(), zkPort,
+                                                           zkLogs.getAbsolutePath());
+      zkProcessBuilder.redirectErrorStream(true);
+      zookeeperProcess = zkProcessBuilder.start();
+      zkIOThread = new Thread(new IOHandlerThread(zookeeperProcess.getInputStream(), System.out));
+      zkIOThread.start();
```

Comment:
Better wrap this logic into a class like `ExternalProcessRunner`. We actually have a generic implementation in https://github.com/caskdata/cdap-apps/blob/develop/TwitterSentiment/src/main/java/co/cask/cdap/apps/flowlet/ExternalProgramExecutor.java

You can modify based on the need (e.g. making it a `ExternalJavaProcessRunner`, so that it constructs the bin/java, classpath,.. etc).